python/cudf/cudf/core/dataframe.py

```diff
@@ -4026,16 +4017,16 @@ def stack(self, level=-1, dropna=True):
         dtype: int64
         """
         assert level in (None, -1)
-        index_as_cols = self.index.to_frame(index=False)._columns
-        new_index_cols = libcudf.filling.repeat(index_as_cols, self.shape[1])
-        [last_index] = libcudf.filling.tile(
-            [column.as_column(self.columns)], self.shape[0]
-        )
-        new_index_cols.append(last_index)
-        index_df = DataFrame(
-            dict(zip(range(0, len(new_index_cols)), new_index_cols))
+        repeated_index = self.index.repeat(self.shape[1])
+        repeated_index = [c for c in repeated_index._columns]
+
+        name_index = DataFrame({0: self._column_names}).tile(self.shape[0])
+        name_index = [c for c in name_index._columns]
+
+        new_index = [*(c for c in repeated_index), name_index[0]]
```

Comment:
`c for c in repeated_index` will iterate element by element through `repeated_index` which would be crazy expensive. What's the goal of what you're trying to do in this line?

Reply:
Uhh, well, it looks confusing, I'll admit, but `repeated_index` is a list of columns. I'm open to reformatting suggestions to make this clearer.

Or am I missing something?

Reply:
I just misread this completely. I think this will simplify things as far as readability:
```suggestion
        new_index = repeated_index + [name_index[0]]
```

Reply:
I've spent too long away from python. :man_facepalming: